### PR TITLE
Improvement: refine CLI commands, arguments, and communication with server documentation 

### PR DESCRIPTION
**Pull request by jhult**
Implements #650.

Many of these commits could (should?) be squashed. I have left them for the moment for ease of review (commit-by-commit).

### CODE REVIEW

File: rust/src/client.rs [hunk 1]
```diff
@@ -14,397 +14,311 @@ use tracing::instrument;
...
+        #[arg(short, long, default_value_t = false)]
+        verbose: bool,
+        /// Output JSON data
+        #[arg(short, long, default_value_t = false)]
+        json: bool,
+    },
+    #[clap(subcommand)]
+    Transactions(Transactions),
+    #[clap(subcommand)]
+    InternalCommands(InternalCommands),
```
Comment (Isaac-DeFrain) [inline rust/src/client.rs]: Should we be consistent with making command names plural vs. singular?

Comment (jhult) [reply rust/src/client.rs]: I think we should be consistent.

Can we use the following or are these really singular?

- StakingLedgers?
- Ledgers?
- Snarks?

**Comment by jhult** [comment]
@Isaac-DeFrain, I know some tests are failing, but I am not sure where I caused the issues. It [looks like on this one](https://buildkite.com/granola/mina-indexer/builds/590#018e7dd4-1dc7-4bb2-8ffe-1fa66cd756a1/36-2529) there was an unexpected end reading from the socket due to additional bytes.

File: rust/src/client.rs [hunk 2]
```diff
@@ -11,400 +11,324 @@ use tokio::{
...
+        #[arg(long)]
+        epoch: u32,
+        /// Network
+        #[arg(long, default_value = "mainnet")]
+        network: String,
+        /// Path to write the staking ledger [default: stdout]
+        #[arg(long)]
+        path: Option<PathBuf>,
+    },
+    /// Query staking ledger by ???
```
Comment (jhult) [inline rust/src/client.rs:217]: @Isaac-DeFrain, can you recommend a better comment here?

Comment (Isaac-DeFrain) [reply rust/src/client.rs:217]: Query staking ledger delegations by epoch?

**Comment by jhult** [comment]
> @Isaac-DeFrain, I know some tests are failing, but I am not sure where I caused the issues. It [looks like on this one](https://buildkite.com/granola/mina-indexer/builds/590#018e7dd4-1dc7-4bb2-8ffe-1fa66cd756a1/36-2529) there was an unexpected end reading from the socket due to additional bytes.

Turns out the buffer size (via `Vec::new()`) wasn't large enough.

**Comment by trevorbernard** [comment]
I like all the structural changes and it simplifies our manual unrolling of the payload. I have some subjective comments that don't affect the validity of this PR. :shipit:

**Review by trevorbernard — APPROVED** [review]

**Pull request merged**